tensorboard/backend/application.py

```diff
@@ -337,6 +342,55 @@ def _create_wsgi_app(self):
     app = _handling_errors(app)
     return app
 
+  @wrappers.Request.application
+  def _serve_plugin_entry(self, request):
+    """Serves a HTML for iframed plugin entry point.
+
+    Args:
+      request: The werkzeug.Request object.
+
+    Returns:
+      A werkzeug.Response object.
+    """
+    name = request.args.get('name')
+    plugins = [
+        plugin for plugin in self._plugins if plugin.plugin_name == name]
+
+    if not plugins:
+      return http_util.Respond(
+          request, 'Plugin not found', 'text/plain', code=404)
+
+    if len(plugins) > 1:
+      # Technically is not possible as plugin names are unique and is checked
+      # by the check on __init__.
+      return http_util.Respond(
+          request, 'Invariant error', 'text/plain', code=400)
```

Comment:
Should be code=500? You could also just raise an `AssertionError` here,
which will properly communicate a 500 as well as logging this exception,
which we probably want.